### PR TITLE
Rhel7 add snapshot

### PR DESCRIPTION
**Pull request by jkonecny12**
Add snapshot feature support to RHEL 7.

*Resolves: rhbz#1113207*

### CODE REVIEW

File: pykickstart/commands/snapshot.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#
+# Jiri Konecny <jkonecny@redhat.com>
+#
+# Copyright 2005, 2006, 2007, 2008, 2012, 2016 Red Hat, Inc.
```
Comment (vpodzime) [inline pykickstart/commands/snapshot.py]: Should be just 2017, AFAICT.

File: pykickstart/commands/snapshot.py [hunk 2]
```diff
@@ -68,28 +85,60 @@ def __str__(self):
         return retval
 
     def _getParser(self):
+        def when_cb(option, opt_str, value, parser):
+            if value.lower() in self.whenMap:
+                parser.values.ensure_value(option.dest,
+                                           self.whenMap[value.lower()])
```
Comment (vpodzime) [inline pykickstart/commands/snapshot.py]: Wouldn't this raise a ``KeyError`` when parsing the opts?

Comment (jkonecny12) [inline pykickstart/commands/snapshot.py]: Truth is I didn't test that, I copied it from `Autopart` class `type` parameter. And it is there for some time, so I *guess* it should work properly.

File: pykickstart/commands/snapshot.py (hunk 1 above)
Comment (jkonecny12) [inline pykickstart/commands/snapshot.py]: Ohh damn. Now I need to rewrite it completely with 2017 code style... 🤣 
Fixed.

**Comment by jkonecny12** [comment]
@clumens Hi, could this be merged please?

**Comment by jkonecny12** [comment]
It should be ok now.

**Pull request merged**